src/AppInstallerTestExeInstaller/main.cpp

```diff
@@ -1,16 +1,116 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+#include <windows.h>
+#include <winreg.h>
+#include <winerror.h>
 #include <iostream>
 #include <fstream>
 #include <filesystem>
 #include <sstream>
 
-// The installer simply prints all args to an output file
+using namespace std::filesystem;
+
+std::wstring_view registrySubkey = L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\";
+std::wstring_view defaultProductID = L"{A499DD5E-8DC5-4AD2-911A-BCD0263295E9}";
+
+std::wstring GenerateUninstaller() {
+    path tempPath = temp_directory_path();
+    path uninstallerPath = tempPath;
+    uninstallerPath /= "UninstallTestExe.bat";
+
+    std::cout << "Uninstaller located at path: " << uninstallerPath << '\n';
+
+    path uninstallerOutputTextFilePath = tempPath;
+    uninstallerOutputTextFilePath /= "TestExeUninstalled.txt";
+
+    std::ofstream uninstallerScript(uninstallerPath);
+    uninstallerScript << "@echo off\n";
+    uninstallerScript << "ECHO. >" << uninstallerOutputTextFilePath << "\n";
+    uninstallerScript << "ECHO AppInstallerTestExeInstaller.exe uninstalled successfully.\n";
+    uninstallerScript.close();
+
+    return uninstallerPath.wstring();
+}
+
+void WriteToUninstallRegistry(std::wstring productID, std::wstring uninstallerPath) {

```

Comment:
>std::wstring [](start = 30, length = 12)

const std::wstring&

Reply:
changed parameter to const std::wstring&